lib/routes/content.js

```diff
@@ -187,27 +188,26 @@ const routes = application => {
      * @param {object} next - Express callback
     */
     app.get(`/${config.content.base}/:type/:id/:revision/${config.content.actions.edit}`, (req, res, next) => {
-      const errors = _.get(req.session, `form.content.add[${req.params.type.toLowerCase()}].errors`, {});
-      const idSess = _.get(req.session, `form.content.add[${req.params.type.toLowerCase()}].id`, {});
-      const revisionSess = _.get(req.session, `form.content.add[${req.params.type.toLowerCase()}].revision`, {});
-      const values = _.get(req.session, `form.content.add[${req.params.type.toLowerCase()}].content`, {});
+      const errors = _.get(req.session, `form.content.edit[${req.params.type.toLowerCase()}].errors`, {});
+      const id = (_.get(req.session, `form.content.edit[${req.params.type.toLowerCase()}].id`) || _.get(req, 'params.id'));
```

Comment:
1. default fallback for both was an object, which was causing errors. `revision` and `id` must be strings
2. this selection chooses to _either_ grab the existing revision id from the session (in the case of a validation/client error) _or_ grabs the actual revision id from the url

Values used below, `_.set(req.session, 'form.content.edit', {` to re-set the session vars.

These are _only_ used in the case of `values` not being empty - ie after an error. Otherwise, we grab them from the url. (starting line 236)